1.3.0 Adaptation

zumbak04 opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Moved and adapted the mod to 1.3.0
- Vrykul use all the viking features like:
   - Innovations
   - Migrations
   - Adventurers
   - Berserks
   - Runestones
   - Blots
   - Hairstyles, beards, backgrounds
   - Family Legacies

# Notes:
- Building got `defender_holding_advantage` instead of `defender_advantage`.
- `development_growth_factor` moved to `county_modifier` instead of `province_modifier`.
- Hairstyle and beard genes are gone.

# Tasks
<details>
<summary>Click to expand</summary>

- [x] Replaced viking cultures and religions
- [x] Allow vrykul to use custom viking religion icons
- [x] Adapt new hair genes
- [x] Check gene indexes
- [x] Adapt eyes and eyelashes
- [x] Adapt cultural hairstyles and beards
- [x] Adapt `fp1_yearly.0052` text
- [x] Check how `fp1_scandinavian_adventurers.0004`, `fp1_scandinavian_adventurers.0011`, `fp1_scandinavian_adventurers.0021` works
- [x] Adapt `on_title_lost` adventurers
- [x] Bring back accepting local traditions?
- [x] Keep Jomsvikings?
- [x] Fix errors mentioning `varangian`
- [x] Relocalized `settled_norse_dynasty_modifier`.
- [x] Check regions where `winter_soldier` is given
- [x] Check you can't offer concubine of another race
- [x] Get rid of Danelaw
- [x] Test that vrykul use viking CBs and raids
- [x] Check I didn't revert `tenet_warmonger`.
- [x] Check murlocs have no hairstyles in the Barbershop.
- [x] Vrykul are shadowed on the main screen
- [x] Fix all errors mentioning:
  - [x] Culture
  - [x] Faith
  - [x] Religion
  - [x] Region
  - [x] Title
- [x] Adapt viking MaAs
- [x] Better not use `north_europe_like_region_trigger` in viking events and create new trigger instead.
- [x] Adapt troll text
- [x] Adapt `fp1_yearly.1001`
- [x] Adapt year triggers
- [x] Fix religion icons
- [x] Fix these hairs, fix `pdxmeshtype.cpp:901`.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/111775050-aef49280-88c9-11eb-826f-236476be8b89.png)

</details>

- [x] Let vrykul send adventurers before High Medieval
- [x] Adapt MaAs icons
- [x] Fix everyone having special eyes and eyebrows
- [ ] Modding coop icon stopped moving.
- [x] Fix errors mentioning `decision`.
- [x] Fix errors mentioning `fp1`.
- [x] Fix `princely_elective_succession_law`
- [x] Fix `embraced_local_traditions_modifier` not having name.
- [x] Check there are no new cultures, religions added.
- [x] Fix opinion errors
- [x] Fix `jomini_script_system.cpp:169`
- [x] Fix `hair` and `beard` errors.
- [x] Replace `has_graphical_culture` by triggers

</details>

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
Play vrykul and test all adapted features.

Test all our ethnicities. Hairs and beards got reworked, so I had to rework the hairs and beards of the races.

@Warcraft-GoA-Development-Team/coders review the Files changed if you can.

TheDrPibb1999 commented:
a majority of the features looked great to me, only things that caught me were;

1, the loc for the huscarls still mention germans 
![Capture](https://user-images.githubusercontent.com/79172797/113375583-930adb00-9324-11eb-9eeb-8d5e95be3651.JPG)

and 2, i got this weird texture when I was ruling over land as the vrykul during an independence rebellion 
![Capture 2](https://user-images.githubusercontent.com/79172797/113375616-a5851480-9324-11eb-91f9-1e3a7130f8e8.JPG)

blots, rune stones, adventures, and all hairstyles/beards looked good to me.

TheGroxEmpire reviewed (commented):
Playing as Odynic Vrykul in Northrend I could not find these features:

1. Blot decision
2. Runestones
3. New Family Legacies
4. Migration CB
5. Runestones decision
6. Adventurers

I can confirm these features are applied:
1. Innovations (Longships)
2. Special MaA (Huscarl only)
3. New hairstyle and beard

Some of the features are bugged:
- Norse background doesn't apply in most screen except here in find spouse / marriage screen:
<details>

![image](https://user-images.githubusercontent.com/47816228/113421483-64106a00-93f5-11eb-828c-e568ff89925b.png)
</details>

- One time I got the event popup to choose my deity. But I'm stuck with that and cannot replace my deity like in base Northern Lord. The other time the event didn't fire at all, leaving my character without chosen deity.

zumbak04 commented:
> 1, the loc for the huscarls still mention germans
> ![Capture](https://user-images.githubusercontent.com/79172797/113375583-930adb00-9324-11eb-9eeb-8d5e95be3651.JPG)

Fixed in 5ea6246730bab2f951f2ab39684cf16d69d48986.

> and 2, i got this weird texture when I was ruling over land as the vrykul during an independence rebellion
> ![Capture 2](https://user-images.githubusercontent.com/79172797/113375616-a5851480-9324-11eb-91f9-1e3a7130f8e8.JPG)

That's due to the Debug mode.

> * One time I got the event popup to choose my deity. But I'm stuck with that and cannot replace my deity like in base Northern Lord. The other time the event didn't fire at all, leaving my character without chosen deity.

What event?

TheGroxEmpire commented on localization/replace/english/wc_mottos_l_english.yml:
```diff
@@ -0,0 +1,43 @@
+ motto_odins_eye:0 "Odyn's One Eye"
+ motto_odins_ravens:0 "Odyn's Ravens"
+ motto_friggs_wisdom:0 "Eonar's Wisdom"
+ motto_friggs_loins:0 "Eonar's Bounteous Loins"
```
> What event?

I didn't screenshot the event but it is similar to Determine Personal Deity from base Northern Lords, here's the loc for it.

zumbak04 replied:
> > What event?
> 
> I didn't screenshot the event but it is similar to Determine Personal Deity from base Northern Lords, here's the loc for it.

This localization is used in the mottos.
I checked and couldn't find working entrances of the game assigning Norse gods. That's why I asked about the event.

TheGroxEmpire replied:
I tried to search for an event ID but the best I got is another loc for it on ``decision_l_english.yml``:
```
select_personal_deity_germanic_decision:0 "Determine Personal Deity"
 select_personal_deity_germanic_decision_tooltip:0 "Think about [ROOT.Char.GetFaith.PantheonTerm]"
 select_personal_deity_germanic_decision_desc:1 "#F It is not uncommon for [ROOT.Char.GetFaith.GetAdherentNamePluralNoTooltip] to single out one amongst [ROOT.Char.GetFaith.PantheonTerm] who suits their particular tastes and aims in life for special attention.\n\nI should contemplate these divine beings and decide which one calls to me the most.#!"
 select_personal_deity_germanic_decision_confirm:0 "Dedicate your Worship to a particular God"
 select_personal_deity_germanic_decision_tt:1 "You begin to live your life according to the values of"
 select_personal_deity_germanic_decision.select_odin:0 "Odin, Glad-of-War, the All-Father"
 select_personal_deity_germanic_decision.select_ullr:0 "Ullr, Swiftest on Snow"
 select_personal_deity_germanic_decision.select_tyr:0 "Tyr, the One Handed"
 select_personal_deity_germanic_decision.select_thor:0 "Thor, Guardian of Man"
 select_personal_deity_germanic_decision.select_freyr:0 "Freyr, Bringer of Peace"
```

zumbak04 replied:
> I tried to search for an event ID but the best I got is another loc for it on `decision_l_english.yml`:

I removed this event before, that's why I surprised you got a Norse god modifier.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/113432785-1264e100-93ef-11eb-8f08-14e901c65017.png)

</details>

TheGroxEmpire replied:
Yeah it appears so that it has been commented. If I find that event again I'll identify the ID.

TheGroxEmpire reviewed (commented):
I also found that Vikings are missing the migration CB (If that's what you meant by it).
But other than that the features seems to be working.

zumbak04 commented:
> I also found that Vikings are missing the migration CB (If that's what you meant by it).

Can't confirm. I see it.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/113425631-476b3680-93e3-11eb-91b8-7b961d21dce9.png)

</details>

TheGroxEmpire commented:
> Can't confirm. I see it.

Yeah, my mistake, apparently I forgot that it was an innovation and my ruler didn't had it

zumbak04 commented:
> Yeah, my mistake, apparently I forgot that it was an innovation and my ruler didn't had it

What culture it was?

TheGroxEmpire commented:
> What culture it was?

It was a random Vrykul in Howling Fjord. It had the innovation just had not been fascinated.

TheGroxEmpire reviewed (commented):
- I'm not sure what happened here but the Shieldmaiden is dueling a non existent character. Accepting it didn't result any new Shieldmaiden:
<details>

![image](https://user-images.githubusercontent.com/47816228/113435973-0dfbf080-940e-11eb-8d82-3eb7aee282e1.png)
```
Event: fp1_shieldmaiden.0001 Character: ONCLICK:CHARACTER,39584 TOOLTIP:CHARACTER,39584 L Erling!!! [39584]
Root: Character Erling Winterskorn of d_winterskorn (Internal ID 39584) [39584]

Saved event targets:
challenged_knight: Character Ran Lowborn of  (Internal ID 39323) [39323]

Saved list targets:
prejudiced_courtiers_list: Character Sveyn Lowborn of  (Internal ID 41383) [41383]Character Dagmar Lowborn of  (Internal ID 34801) [34801]Character Guthlaug Lowborn of  (Internal ID 34824) [34824]Character Katla Lowborn of  (Internal ID 39321) [39321]Character Hjejskard Lowborn of  (Internal ID 39322) [39322]Character Gyda Winterskorn of  (Internal ID 39586) [39586]Character Svanhildr Lowborn of  (Internal ID 39324) [39324]Character Ulfrun Lowborn of  (Internal ID 39325) [39325]Character Thorkell Lowborn of  (Internal ID 39327) [39327]Character Kari Wrynn of  (Internal ID 39583) [39583]Character Andjor Winterskorn of c_ivald (Internal ID 39585) [39585]Character Ghrent Malmqvist of c_rivenwood (Internal ID 39589) [39589]Character Hjejskard Malmqvist of c_skorn (Internal ID 39590) [39590]
Description:
fp1_shieldmaiden.0001.desc
```
</details>

- Hairy custom models:
<details>

![image](https://user-images.githubusercontent.com/47816228/113436342-bf028b00-940e-11eb-9d96-fbe0423848ae.png)
![image](https://user-images.githubusercontent.com/47816228/113436428-e35e6780-940e-11eb-91cc-ea1ab07d078e.png)
![image](https://user-images.githubusercontent.com/47816228/113436455-ef4a2980-940e-11eb-9c2a-9ef49cebcad8.png)

</details>

- The Alliance Union formation event uses the Northern background:
<details>

![image](https://user-images.githubusercontent.com/47816228/113437167-587e6c80-9410-11eb-8bd5-502834de83e0.png)
</details>

zumbak04 commented:
> * I'm not sure what happened here but the Shieldmaiden is dueling a non existent character. Accepting it didn't result any new Shieldmaiden:

I hope you didn't fire it from the console?

zumbak04 commented:
> * I'm not sure what happened here but the Shieldmaiden is dueling a non existent character. Accepting it didn't result any new Shieldmaiden:

Should be fixed in af8a4e5f00d42b4982355380108a469719d4c07e.

> * Hairy custom models:

Fixed in 79027308e20f0641667c436da0dce55dc339765b.

> * The Alliance Union formation event uses the Northern background:

Fixed in af87f02cb8e5027b8ae81f2136b73e06ebf36977.

TheDrPibb1999 reviewed (approved):
Well all issues seem to be addressed, LGTM!

TheGroxEmpire reviewed (commented):
- All of my aforementioned issues were fixed.

- DLC Loading screen sometimes shows up:
<details>

![image](https://user-images.githubusercontent.com/47816228/113470760-260d5780-9482-11eb-8f5e-83e3e170ffc0.png)

</details>

- Horde AI sometimes doesn't create the Horde Union until decades later after the Alliance (On one of my playtest: 583 start > Horde created at 605). On previous branch it was almost immediately. The decision was there and available when I switch character to the Warchief.
<details>

![image](https://user-images.githubusercontent.com/47816228/113470855-c794a900-9482-11eb-9db1-0162790534d1.png)

</details>

zumbak04 commented:
> * DLC Loading screen sometimes shows up:

Should be fixed in 395ee6f19ecc74e33fbb8b8256688a5f7605a782.

> * Horde AI sometimes doesn't create the Horde Union until decades later after the Alliance formation (On one of my playtest: 583 start > Horde created at 605). On previous branch it was almost immediately. The decision was there and available when I switched character to the Warchief.

Should be fixed in 700e8227c4ad443787c003f768503a920ac64365.

arithon reviewed (approved):
loc LGTM

TheGroxEmpire reviewed (approved):
Looks like the bugs were fixed. All good here.
